### DISCUSSION TITLE
[Bug] Auth Token is not refreshed when using plugin with SFDC API

### DISCUSSION THREAD

**Discussion by camsellem** [grafana/grafana-infinity-datasource#1040] [Specific APIs]
## **Describe the bug**

I'm using the Infinity plugin to query Salesforce API. I'm experiencing the following bug - After setting up the datasource and creating my dashboard, everything works just fine. Although if I come back 24/48 hours after on the dashboard I'm getting a `401 unauthorized` error on every panels. To fix it I need to edit the datasource and perform a health check - then it works again. My assumption is that my token expires and it's not refreshed within the dashboard hence the need of editing the datasource again.

## **Steps to reproduce**

Create a connected App in SFDC to access the REST API
Create an Infinity Datasource 
Authenticate using OAuth2 using Client ID and Client Secret
Create a Panel and retrieve some SFDC Data
Open the dashboard again a day or two after
Error should pop up
Edit the datasource and perforce a health check to refresh the token
Dashboard is ok again

**Note:** bugs will be fixed only if enough information provided to reproduce the issue

## **Sample Data**

Provide a [link](https://gist.github.com) for sample data, or provide sample data bellow

## **Screenshots**

![Image](https://github.com/user-attachments/assets/3df02fb8-f074-4fe3-8f37-56d90a5a3653)

## **Version Details:**

- Grafana version : 11.2
- Plugin version : 2.9.6

**Comment by peterneal-eversure** [comment]
I'm running into this problem with the 0Auth2 client Credentials token reference not working, This and the 0Auth2 JWT flow not accepting the private key (always complains about the file format). I can't make a reliable 24/7 connection to Salesforce for a dashboard in a playlist. 

Does anyone know a workaround?
